static/app/utils/useRequest.tsx

```diff
@@ -0,0 +1,254 @@
+import {useState} from 'react';
+
+import {Client} from 'app/api';
+import OrganizationStore from 'app/stores/organizationStore';
+import {useLegacyStore} from 'app/stores/useLegacyStore';
+import parseLinkHeader from 'app/utils/parseLinkHeader';
+import RequestError from 'app/utils/requestError/requestError';
+import useApi from 'app/utils/useApi';
+
+type State = {
+  data: null | any;
+  /**
+   * This is state for when fetching data from API
+   */
+  loading: boolean;
+  /**
+   * The error that occurred if fetching failed
+   */
+  error: null | RequestError;
+  /**
+   * Indicates that Team results (from API) are paginated and there are more
+   * Teams that are not in the initial response.
+   */
+  hasMore: null | boolean;
+  /**
+   * The last query we searched. Used to validate the cursor
+   */
+  lastSearch: null | string;
+  /**
+   * Pagination
+   */
+  nextCursor?: null | string;
+};
+
+export type Result = {
+  /**
+   * This is an action provided to consumers for them to update the current
+   * result set using a simple search query.
+   *
+   * Will always add new options into the store.
+   */
+  onSearch: (searchTerm: string) => Promise<void>;
+  nextPage: () => Promise<void>;
+  get: () => Promise<void>;
+  post: (_) => Promise<void>;
+  update: (_) => Promise<void>;
+  del: () => Promise<void>;
+} & State;
+
+type Options = {
+  /**
+   * Number of repositories to return
```

Comment:
I believe this comment should be generic, unless `useRequest` will only be used for `useRepositories`